### PR TITLE
add IMDb

### PR DESCRIPTION
**Pull request by Chen-Roth**

### CODE REVIEW

File: examples/IMDb/IMDb/config.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import os
+from typing import Dict, Any
+import yaml
+
+def load_od_config() -> Dict[str, Any]:
+    # Load the existing YAML config
+    root = os.path.abspath(os.path.dirname(__file__))
+    file_path = os.path.join(root, 'project_config.yaml')
```
Comment (tomkoren21) [inline examples/IMDb/IMDb/config.py:8]: I think it might make sense to standardise this. I saw on `armbench` a different naming. Is this name the new default?

File: examples/IMDb/IMDb/data/preprocess.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+
+import json
+from keras.preprocessing.text import tokenizer_from_json
+import pandas as pd
+from os.path import join
+
+from IMDb.config import CONFIG
```
Comment (tomkoren21) [inline examples/IMDb/IMDb/data/preprocess.py:7]: Consider lower-casing this folder name

File: examples/IMDb/README.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+#### Unsupervised Clustering
+
+Tensorleap's platform provides an unsupervised clustering of samples based on their similarities.
+
+In our use-case we can see that this clustering technique was able to group together samples with common features. For example, in this Population Exploration plot, each cluster is colored, while the size of the dot represents the number accuracy each sample:
+
+![Population Exploration - Unsupervised Clustering](images/img_4.png)
+
+It can be seen that the left-side groups consist of samples with a known ground truth of "positive".
```
Comment (tomkoren21) [inline examples/IMDb/README.md:48]: Consider rephrasing the end. Positively labelled samples

File: examples/IMDb/README.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+![Population Exploration - Unsupervised Clustering2](images/img_5.png)
+
+#### Sample Analysis
+
+The Tensorleap platform provides us with a way to further explore the model's response to specific data samples.
+
+For example, performing **Sample Analysis** of one of the failing samples shows a **false-positive** prediction made by the model, with a heat-map overlay that scores the significance of each word to the `positive` prediction:
```
Comment (tomkoren21) [inline examples/IMDb/README.md:55]: "failing sample" -> high loss sample

File: examples/IMDb/leap_binder.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+    train_label_size = int(0.9 * CONFIG['NUMBER_OF_SAMPLES'] / 2)
+    val_label_size = int(0.1 * CONFIG['NUMBER_OF_SAMPLES'] / 2)
+    df = df[df['subset'] == 'train']
+    train_df = pd.concat([df[df['gt'] == 'pos'][:train_label_size], df[df['gt'] == 'neg'][:train_label_size]],
+                         ignore_index=True)
+    val_df = pd.concat([df[df['gt'] == 'pos'][train_label_size:train_label_size + val_label_size],
+                        df[df['gt'] == 'neg'][train_label_size:train_label_size + val_label_size]], ignore_index=True)
+    ohe = {"pos": [0., 1.0], "neg": [1.0, 0.]}
+
+    train = PreprocessResponse(length=2 * train_label_size, data={"df": train_df, "tokenizer": tokenizer, "ohe": ohe})
```
Comment (tomkoren21) [inline examples/IMDb/leap_binder.py:36]: Maybe limit the train_label_size and val_label_size by len(df['gt'] == LABEL)? If someone places a number of samples which is greater than df size we might have a bug here since length would be incorrect

File: examples/IMDb/leap_custom_test.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+from keras.losses import BinaryCrossentropy
+
+def check_custom_test():
+    print("started custom tests")
+    responses = preprocess_func()
+    train = responses[0]
+    val = responses[1]
+    responses_set = train
+    dir_path = os.path.dirname(os.path.abspath(__file__))
+    model_path = 'model/fabriceyhc-bert-base-uncased-imdb.onnx'
```
Comment (tomkoren21) [inline examples/IMDb/leap_custom_test.py:15]: is this onnx a part of the repo? if not maybe call cloud here to get it?

File: examples/IMDb/leap_custom_test.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    print("finish tests")
+
+def check_custom_test_dense():
+    print("started custom tests")
+    responses = preprocess_func()
+    train = responses[0]
+    val = responses[1]
+    responses_set = train
+    dir_path = os.path.dirname(os.path.abspath(__file__))
+    model_path = 'model/exported-model.h5'
```
Comment (tomkoren21) [inline examples/IMDb/leap_custom_test.py:74]: if this should be used then call cloud here?

File: examples/IMDb/leap_custom_test.py [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        else:
+            text.append("neg")
+        print(f'gt: {text[0]}')
+
+    print("finish tests")
+
+
+if __name__ == '__main__':
+    check_custom_test()
+    # check_custom_test_dense()
```
Comment (tomkoren21) [inline examples/IMDb/leap_custom_test.py:121]: why not run both?

**Review by tomkoren21 — APPROVED** [review]
LGTM

File: examples/IMDb/IMDb/config.py (hunk 1 above)
Comment (Chen-Roth) [reply examples/IMDb/IMDb/config.py:8]: changing name to "load_config"

File: examples/IMDb/IMDb/data/preprocess.py (hunk 2 above)
Comment (Chen-Roth) [reply examples/IMDb/IMDb/data/preprocess.py:7]: changed to "imdb"

File: examples/IMDb/IMDb/gcs_utils.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def _download(cloud_file_path: str, local_file_path: Optional[str] = None) -> str:
+    """
+    Downloads a file from Google Cloud Storage to a local directory, ensuring it exists locally for further use.
+    :param cloud_file_path: The path to the file in Google Cloud Storage.
+    :param local_file_path:  The optional local path to save the downloaded file.
+    :return:The path to the downloaded local file.
+    """
+    BASE_PATH = "imdb"
```
[the remark this answers is not in the document]
Comment (Chen-Roth) [reply examples/IMDb/IMDb/gcs_utils.py:25]: moved to config file

File: examples/IMDb/README.md (hunk 3 above)
Comment (Chen-Roth) [reply examples/IMDb/README.md:48]: has changed

File: examples/IMDb/README.md (hunk 4 above)
Comment (Chen-Roth) [reply examples/IMDb/README.md:55]: Done

File: examples/IMDb/leap_custom_test.py (hunk 8 above)
Comment (Chen-Roth) [reply examples/IMDb/leap_custom_test.py:121]: to run them both need to change some code. The second test was deleted

File: examples/IMDb/leap_binder.py (hunk 5 above)
Comment (Chen-Roth) [reply examples/IMDb/leap_binder.py:36]: change to len(df)

File: examples/IMDb/leap_custom_test.py (hunk 6 above)
Comment (Chen-Roth) [reply examples/IMDb/leap_custom_test.py:15]: uploaded to the repo

File: examples/IMDb/leap_custom_test.py (hunk 7 above)
Comment (Chen-Roth) [reply examples/IMDb/leap_custom_test.py:74]: uploaded to the repo
